### PR TITLE
[Symfony 3.x] Fixes for Symfony 3 compatibility

### PR DESCRIPTION
**Pull request by andrerom**
review ping @bdunogier @alongosz @Nattfarinn

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/FragmentPass.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function process(ContainerBuilder $container)
         $fragmentListenerDef = $container->findDefinition('fragment.listener');
         $fragmentListenerDef
             ->setFactory([new Reference('ezpublish.fragment_listener.factory'), 'buildFragmentListener'])
-            ->addArgument('%fragment.listener.class%');
+            ->addArgument('Symfony\Component\HttpKernel\EventListener\FragmentListener');
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/FragmentPass.php]: `FragmentListener::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/RouterPass.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('router.default')) {
+            return;
+        }
+
+        $container
+            ->findDefinition('router.default')
+            ->setClass('eZ\Bundle\EzPublishCoreBundle\Routing\DefaultRouter');
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/RouterPass.php]: `::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 3]
```diff
@@ -26,15 +26,22 @@ public function process(ContainerBuilder $container)
 
         $configResolverRef = new Reference('ezpublish.config.resolver');
         $repositoryReference = new Reference('ezpublish.api.repository');
-        // Inject the Repository in the authentication provider.
+
+        // Override and inject the Repository in the authentication provider.
         // We need it for checking user credentials
         $daoAuthenticationProviderDef = $container->findDefinition('security.authentication.provider.dao');
+        $daoAuthenticationProviderDef->setClass(
+            'eZ\Publish\Core\MVC\Symfony\Security\Authentication\RepositoryAuthenticationProvider'
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]: `::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 4]
```diff
@@ -26,15 +26,22 @@ public function process(ContainerBuilder $container)
...
+        );
         $daoAuthenticationProviderDef->addMethodCall(
             'setRepository',
             array($repositoryReference)
         );
 
         $anonymousAuthenticationProviderDef = $container->findDefinition('security.authentication.provider.anonymous');
+        $anonymousAuthenticationProviderDef->setClass(
+            'eZ\Publish\Core\MVC\Symfony\Security\Authentication\AnonymousAuthenticationProvider'
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]: `::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 5]
```diff
@@ -60,6 +68,9 @@ public function process(ContainerBuilder $container)
         }
 
         $successHandlerDef = $container->getDefinition('security.authentication.success_handler');
+        $successHandlerDef->setClass(
+            'eZ\Publish\Core\MVC\Symfony\Security\Authentication\DefaultAuthenticationSuccessHandler'
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]: `::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 6]
```diff
@@ -50,6 +57,7 @@ public function process(ContainerBuilder $container)
         }
 
         $httpUtilsDef = $container->findDefinition('security.http_utils');
+        $httpUtilsDef->setClass('eZ\Publish\Core\MVC\Symfony\Security\HttpUtils');
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]: `::class` ?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php [hunk 7]
```diff
@@ -25,6 +25,7 @@ public function process(ContainerBuilder $container)
 
         $assetFactoryDef = $container->findDefinition('assetic.asset_factory');
         $assetFactoryDef
+            ->setClass('eZ\Bundle\EzPublishCoreBundle\Assetic\AssetFactory')
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php]: `::class` ?

Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php]: Would you care explaining why we need to _add_ this `setClass()` call ? I would have understood if we had replaced a `setClass('%something.class%')` or something related to the commit message, but not this.

Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php]: See yml changes

Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php]: Specifically templating.yml

**Comment by andrerom** [comment]
> Side note: maybe for new code everywhere (including class mocks) we should use ::class (provided composer requires php>=5.5). It makes maintenance / refactoring easier.

yes, please do that on all new code and code refactorings, but be careful on code that will need backports to 5.4.

**Comment by andrerom** [comment]
Fixed, ping @bdunogier

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php (hunk 7 above)
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/AsseticPass.php]: So we were replacing the class for some services by overriding the class parameter, and we are switching to `setClass`. Alright.

**Comment by bdunogier** [comment]
Merged.

Is there a JIRA issue linked to this ? There should be one, since we have an Epic for Sf3.

**Comment by andrerom** [comment]
Could be, but atm there might still be issues until we also setup functional tests with Symfony 3 to verify, so maybe that is rather the story we need to create in the epic.

**Comment by emodric** [comment]
It would be great if this was backported to LTS :)

**Comment by andrerom** [comment]
That won't help you much, fixes in other repos where not done in 1.7 either. Once we get a full overview over sf3 fixes and know they don't have side effects we can consider doing that, but these are hardly stable fixes yet.

**Comment by emodric** [comment]
Sure, I'm aware of that, but the point is to be able to at least install Sf 3 and start playing with it.

**Comment by andrerom** [comment]
If you want to play, [go play](https://github.com/andrerom/ezplatform/tree/2.0) ;)

**Comment by emodric** [comment]
> If you want to play, go play ;)

Uuu, mised that one ;)

**Pull request merged**